pkg/controller/agentregistration/server.go

```diff
@@ -45,17 +50,35 @@ func RunAgentRegistrationServer(ctx context.Context, port int, clientHolder *hel
 		}
 	})))
 
+	// example path: /agent-registration/manifests/cluster1?klusterletconfig=default
 	mux.Handle("/agent-registration/manifests/", authMiddleware(clientHolder, http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		var err error
 		urlparams := strings.Split(r.RequestURI, "/")
 		if len(urlparams) < 4 {
 			http.Error(w, fmt.Sprintf("Invalid request: %s", r.RequestURI), http.StatusBadRequest)
 			return
 		}
 
-		clusterID := urlparams[3]
+		klusterletClusterAnnotations := map[string]string{
+			"agent.open-cluster-management.io/create-with-default-klusterletaddonconfig": "true",
+		}
+
+		// Get KlusterletConfig
+		var kc *klusterletconfigv1alpha1.KlusterletConfig
+		klusterletconfigName := r.URL.Query().Get("klusterletconfig")
+		if klusterletconfigName != "" {
+			kc, err = klusterletconfigLister.Get(klusterletconfigName)
+			if err != nil && !apierrors.IsNotFound(err) {
+				http.Error(w, err.Error(), http.StatusInternalServerError)
+				return
+			}
+
+			klusterletClusterAnnotations["agent.open-cluster-management.io/klusterlet-config"] = klusterletconfigName
```

Comment:
All places use `agent.open-cluster-management.io/klusterlet-config` will be replaced with a constant value after this PR merged: https://github.com/stolostron/cluster-lifecycle-api/pull/18

Reply:
Except for calling `WithKlusterletConfig(kc)`, agent-registration also needs to add this annotation to `klusterletClusterAnnotations` to make sure agent-created managedcluster CR has this annotation. 

So that in the future reconcile loop in manifestwork-controller, the manifests keeps the same.